### DISCUSSION TITLE
Should we skip downsampling before running PCA if we have low number of cells to start with?

### DISCUSSION THREAD

**Discussion by denvercal1234GitHub** [ImmuneDynamics/Spectre#99] [Q&A]
Hi there, 

I have quite a small number of cell per sample (below); some samples have much less cells than others. Is it a "bad" bias to not do downsampling before running dimensionality reduction? 

Thank you for your guidance!

<img width="81" alt="Screen Shot 2021-09-02 at 3 41 48 AM" src="https://user-images.githubusercontent.com/62179082/131772881-697fc7cc-1a21-4482-bb78-734dc78bba50.png">

**Comment by tomashhurst** [comment]
@denvercal1234GitHub It is ideal to either have equivalent numbers of cells from each sample/group in the plots (we usually do this with blood/PBMCs), or to downsample in a group/sample-weighted method (e.g. what we do with our CNS data). However, in situations where the numbers of cells in some files is extremely low, it might be very difficult to downsample in a _truthful_ way and maintain enough to see small populations. If this is the case, then it is OK to just use what you have -- just make sure to describe clearly in your figure legend and/or methods, so that differences in cell numbers on the plot are not interpreted to be reflective of what is happening in the actual sample. In an ideal world there would be plenty of cells in each sample, but that's not always realistic, and this approach allows you to manage that. 

For example, in a [recent COVID-19 paper](https://www.researchsquare.com/article/rs-802084/v1) we did both -- one panel we downsample in a group-weighted sense because we had enough cells (so differences in the number of cells on the plot represent the difference in cells/uL), but in another we didn't because of low cell numbers (so we just used what we had).

**Reply by denvercal1234GitHub** [reply]
Thank you very much @tomashhurst! The reason is because when I did not downsample and set meta.k = auto, I obtained the clusters shown below, which looks horrible lol - This could ofc be because they are not distinct populations or that the markers I chose for clustering were not enough to separate them or that number of cells are not large enough. Though, do you have any thoughts to improve it, e.g., setting lower manually meta.k? 

Also, is there a way to highlight cells of a particular cluster and dim the rest of the other cell clusters in UMAP? 

Thank you again for your prompt responses!

<img width="223" alt="Screen Shot 2021-09-02 at 4 40 28 AM" src="https://user-images.githubusercontent.com/62179082/131778216-f5ecdff9-ff6f-48dc-893e-ade78cc1007d.png">
